python/sglang/srt/managers/data_parallel_controller.py

```diff
@@ -583,6 +571,19 @@ def run_data_parallel_controller_process(
     kill_itself_when_parent_died()
     parent_process = psutil.Process().parent()
 
+    # Set up SIGQUIT handler to propagate to parent
+    # This ensures that if a scheduler sends SIGQUIT to DPC, we propagate it to Engine
+    def sigquit_handler(signum, frame):
+        logger.error(
+            "DataParallelController received SIGQUIT from child. "
+            "Propagating to parent Engine process."
+        )
+        parent_process.send_signal(signal.SIGQUIT)
+        # Use os._exit to avoid running any cleanup that might hang
+        os._exit(1)
```

Comment:
Should we remove this line?
This is because when the Engine gets SIGQUIT, it will call `kill_process_tree` to clean up all the child processes. If we remove this line, the `kill_process_tree` can correctly finish the cleaning.
If we keep this line, `kill_process_tree` won't be able to kill all scheduler processes because DPC is dead and engine process cannot find the child scheduler processes anymore.